### PR TITLE
New sample: Generate elevation profile with Local Server

### PR DESCRIPTION
**Pull request by duffh**
# Description

Added a new sample to demonstrate how to create an elevation profile using a geoprocessing package executed with ArcGIS Runtime Local Server.

## Type of change

- [x] New sample implementation

**Platforms tested on**:

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI

## Checklist

- [x] Runs and compiles on all active platforms
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] `sample_sync.py` runs without making changes
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] Code is commented with correct formatting
- [x] All variable and method names are good and make sense
- [x] There is no leftover commented code
- [x] Screenshots are correct size and display in description tab

### CODE REVIEW

File: tools/CI/README_Metadata_StyleCheck/README_style_checker.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     'ArcGIS Pro',
     'GeoPackage',
     'loadStatus',
+    'Local Server',
```
Comment (ZackAllen) [inline tools/CI/README_Metadata_StyleCheck/README_style_checker.py:20]: TBD change on this checker script.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+﻿<UserControl x:Class="ArcGISRuntime.WPF.Samples.LocalServerGenerateElevationProfile.LocalServerGenerateElevationProfile"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:esri="http://schemas.esri.com/arcgis/runtime/2013">
+    <Grid>
+        <Grid.ColumnDefinitions>
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml:6]: All of the Column properties are not needed when using the `BorderStyle`.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+        private GeoprocessingTask _gpTask;
+        private GeoprocessingJob _gpJob;
+        private Raster _arranRaster;
+        private LocalGeoprocessingService _localGPService;
+        private GraphicsOverlay _graphicsOverlay;
+        private GraphicsOverlay _tempGraphicsOverlay;
+        private Viewpoint _rasterExtentViewPoint;
+        private RasterLayer _rasterLayer;
+        private SpatialReference _rasterLayerSpatialReference;
+        private Esri.ArcGISRuntime.Geometry.PointCollection _pointCollection;
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs:64]:
```suggestion
        private PointCollection _pointCollection;
```

**Review by ZackAllen — CHANGES_REQUESTED** [review]

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+            if (_gpJob != null)
+            {
+                _ = _gpJob.CancelAsync();
+
+                // Remove the event handlers.
+                _gpJob.ProgressChanged -= GpJob_ProgressChanged;
+                _gpJob.StatusChanged -= GpJob_StatusChanged;
+
+                // Update the UI.
+                MyCancelJobButton.Visibility = Visibility.Collapsed;
```
[the remark this answers is not in the document]
Comment (duffh) [reply src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs:563]: I prefer the longer form as I think it's more obvious to the user but I'd be happy to swap it over if you think it's unnecessary.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,577 @@
...
+            }
+        }
+
+        private async Task GeoViewTappedTask(GeoViewInputEventArgs e)
+        {
+            // Get the location from the tapped screen position.
+            MapPoint point = await MySceneView.ScreenToLocationAsync(e.Position);
+
+            // Project the point to the spatial reference of the raster layer.
+            MapPoint projectedPoint = (MapPoint)GeometryEngine.Project(point, _rasterLayerSpatialReference);
```
[the remark this answers is not in the document]
Comment (duffh) [reply src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs]: Added a try catch for error output and added a check for a valid `SpatialReference` on click event.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,577 @@
...
+                MessageBox.Show($"Please ensure that local server {localServerVersion.FileVersion} is installed prior to using the sample. The download link is in the description. Message: {ex.Message}", "Local Server failed to start");
+                return;
+            }
+
+            // Create a scene with a topographic basemap style.
+            MySceneView.Scene = new Scene(BasemapStyle.ArcGISHillshadeDark);
+
+            // Create a new raster from a local file and display it on the scene.
+            _arranRaster = new Raster(_rasterPath);
+            await DisplayRaster(_arranRaster);
```
[the remark this answers is not in the document]
Comment (ZackAllen) [reply src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs:119]: Awaiting DisplayRaster without a try/catch in our discard would throw an exception that never gets handled properly. I would also move `await DisplayRaster` into the try/catch.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,598 @@
...
+
+            // Get the polyline's center point coordinates.
+            MapPoint centerPoint = _polyline.Extent.GetCenter();
+            double centerX = centerPoint.X;
+            double centerY = centerPoint.Y;
+
+            // Calculate the position of a point perpendicular to the centre of the polyline.
+            double lengthX = endPointX - centerX;
+            double lengthY = endPointY - centerY;
+            MapPoint cameraPositionPoint = new MapPoint(centerX + lengthY * 2, centerY - lengthX * 2, 1200,
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/LocalServer/LocalServerGenerateElevationProfile/LocalServerGenerateElevationProfile.xaml.cs:363]:
```suggestion
            var cameraPositionPoint = new MapPoint(centerX + lengthY * 2, centerY - lengthX * 2, 1200,
```
IN our style we can use `var` when it precedes a constructor.

**Review by ZackAllen — APPROVED** [review]

**Pull request merged**